genesis/process/sovereignGenesisBlockCreator.go

```diff
@@ -0,0 +1,286 @@
+package process
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/factory"
+	"github.com/multiversx/mx-chain-go/genesis"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/update"
+	"github.com/multiversx/mx-chain-go/vm"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type sovereignGenesisBlockCreator struct {
+	*genesisBlockCreator
+}
+
+// NewSovereignGenesisBlockCreator creates a new sovereign genesis block creator instance
+func NewSovereignGenesisBlockCreator(gbc *genesisBlockCreator) (*sovereignGenesisBlockCreator, error) {
+	if gbc == nil {
+		return nil, errNilGenesisBlockCreator
+	}
+
+	return &sovereignGenesisBlockCreator{
+		genesisBlockCreator: gbc,
+	}, nil
+}
+
+// CreateGenesisBlocks will create sovereign genesis blocks
+func (gbc *sovereignGenesisBlockCreator) CreateGenesisBlocks() (map[uint32]data.HeaderHandler, error) {
+	if !mustDoGenesisProcess(gbc.arg) {
+		return gbc.createSovereignEmptyGenesisBlocks()
+	}
+
+	if mustDoHardForkImportProcess(gbc.arg) {
+		err := gbc.arg.importHandler.ImportAll()
+		if err != nil {
+			return nil, err
+		}
+
+		err = gbc.computeSovereignDNSAddresses(gbc.arg.EpochConfig.EnableEpochs)
```

Comment:
do not compute DNS addresses, just simply set it as the new dnsv2 address.

Reply:
`computeSovereignDNSAddresses` calls `computeDNSAddresses`, which adds the sc address in blockchainHook:

```go
		scResultingAddress, errNewAddress := blockChainHook.NewAddress(address, accountStartNonce, dnsSC.VmTypeBytes())
		if errNewAddress != nil {
			return errNewAddress
		}

		dnsSC.AddAddressBytes(scResultingAddress)

		encodedSCResultingAddress, err := gbc.arg.Core.AddressPubKeyConverter().Encode(scResultingAddress)
		if err != nil {
			return err
		}
		dnsSC.AddAddress(encodedSCResultingAddress)
```

Isn't that ok? Or is there anything else I should change/add?

Reply:
The name with `compute` is kept for consistency with the one from original component

Reply:
ok